orm/src/main/java/org/hibernate/search/hcore/impl/SearchFactoryReference.java

```diff
@@ -22,16 +20,16 @@
  */
 public final class SearchFactoryReference implements Service {
 
-	private volatile CompletableFuture<ExtendedSearchIntegrator> extendedIntegratorFuture;
+	private volatile ExtendedSearchIntegrator extendedIntegrator;
 
-	public void initialize(CompletableFuture<ExtendedSearchIntegrator> future) {
-		extendedIntegratorFuture = future;
+	public void initialize(ExtendedSearchIntegrator extendedIntegrator) {
+		this.extendedIntegrator = extendedIntegrator;
 	}
 
 	public ExtendedSearchIntegrator getSearchIntegrator() {
-		final CompletableFuture<ExtendedSearchIntegrator> value = this.extendedIntegratorFuture;
+		ExtendedSearchIntegrator value = this.extendedIntegrator;
 		if ( value != null ) {
-			return value.join();
+			return value;
```

Comment:
If I understand correctly, this does not just "avoid redundant delegation".

This has two side effects. We'll take CDI as an example:

1. Before this change, if someone tried to use Hibernate Search during CDI beans initialization, the call would block because Hibernate Search wasn't initialized yet, which would result in a deadlock since Hibernate Search waits for CDI to be initialized. After this change, such calls will simply fail. So in that respect, I guess this change is a good thing.
2. Before this change, if someone tried to use Hibernate Search *just after* CDI beans initialization, before Hibernate Search had the time to fully initialize, the call would block, and resume once Hibernate Search is fully initialized. After this change, such calls will simply fail. And **that** is not a good thing...

So in the end, I'd rather add a (configurable) timeout to the `join()`, so we can get the benefits of your patch (1) without its downsides (2).